src/control/lib/hardware/sysfs/provider.go

```diff
@@ -95,6 +128,10 @@ func (s *Provider) GetTopology(ctx context.Context) (*hardware.Topology, error)
 		return nil, err
 	}
 
+	if err := s.addNetvscDevices(topo); err != nil {
```

Comment:
Given this is just a different type of virtual network device, isn't it worth adding into `addVirtualNetDevices()` rather than creating a separate function?

Reply:
At the end, it is not just a standard Virtual Device as it does not appear the standard virtual device directory. Indeed, it is managed in /sys more or less as a standard PCI device: Microsoft Windows magic.
This is why, I decide to manage it in a different function to clearly identify this unusual Virtual device.
However, I have no strong opposition to merge this code in `addVirtualNetDevices()` function.
@kjacque what is your opinion on this point.

Reply:
I'd be inclined to integrate it with `addPCIDevices`, since the new devices are in the same part of the tree, and walking sysfs is a bit costly as it is. `addPCIDevices` should be renamed to `addSubsystemDevices` or something like that, to avoid a misleading name, as they will no longer be only PCI devices.

Integrating with that function will also solve the problem of duplicating code.

Reply:
- [ ] Merge `addPCIDevices()` and `addNetvscDevices()` into a new function `addSubsystemDevices()`